### PR TITLE
feat: Update substrate to gear-polkadot-stable2409

### PR DESCRIPTION
**Pull request by ukint-vs**
- Remove native executor
- Add metadata hash extension https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/guides/enable_metadata_hash/index.html
- update `wasm-builder`
- use new construct runtime macro
- update `jsonrpsee`

@reviewer-or-team

### CODE REVIEW

File: node/service/src/rpc/mod.rs [hunk 1]
```diff
@@ -162,24 +158,14 @@ where
...
     } = gear;
 
-    let chain_name = chain_spec.name().to_string();
-    let genesis_hash = client
-        .block_hash(0)
-        .ok()
-        .flatten()
-        .expect("Genesis block exists; qed");
-    let properties = chain_spec.properties();
-    io.merge(ChainSpec::new(chain_name, genesis_hash, properties).into_rpc())?;
```
Comment (breathx) [inline node/service/src/rpc/mod.rs:172]: chain spec io?

File: pallets/gear-builtin/rpc/src/lib.rs [hunk 2]
```diff
@@ -80,8 +80,9 @@ where
         let api = self.client.runtime_api();
         let best_hash = self.client.info().best_hash;
 
-        fn map_err(err: impl std::fmt::Debug, desc: &'static str) -> JsonRpseeError {
-            CallError::Custom(ErrorObject::owned(8000, desc, Some(format!("{err:?}")))).into()
+        fn map_err(error: impl ToString, desc: &'static str) -> ErrorObjectOwned {
+            ErrorObject::owned(Error::RuntimeError.into(), desc, Some(error.to_string()))
+            // CallError::Custom(ErrorObject::owned(8000, desc, Some(format!("{err:?}")))).into()
```
Comment (breathx) [inline pallets/gear-builtin/rpc/src/lib.rs:85]: deadcode

File: runtime/common/src/apis.rs [hunk 3]
```diff
@@ -31,7 +31,7 @@ macro_rules! impl_runtime_apis_plus_common {
 					Executive::execute_block(block);
 				}
 
-				fn initialize_block(header: &<Block as BlockT>::Header) {
+				fn initialize_block(header: &<Block as BlockT>::Header) -> sp_runtime::ExtrinsicInclusionMode {
```
Comment (breathx) [inline runtime/common/src/apis.rs:34]: seems like breaking change for nodes?

File: runtime/vara/src/integration_tests.rs [hunk 4]
```diff
@@ -543,91 +544,6 @@ fn dust_ends_up_in_offset_pool() {
         });
 }
 
-#[test]
```
Comment (breathx) [inline runtime/vara/src/integration_tests.rs:546]: why?

File: runtime/vara/src/lib.rs [hunk 5]
```diff
@@ -1293,130 +1307,325 @@ impl pallet_vesting::Config for Runtime {
...
+        RuntimeFreezeReason,
+        RuntimeHoldReason,
+        RuntimeSlashReason,
+        RuntimeLockId,
+        RuntimeTask
+    )]
+    pub struct Runtime;
+
+    #[runtime::pallet_index(0)]
+    pub type System = frame_system::Pallet<Runtime>;
```
Comment (breathx) [inline runtime/vara/src/lib.rs:1333]: and similar?
```suggestion
    pub type System = frame_system;
```

File: runtime/vara/src/migrations.rs [hunk 6]
```diff
@@ -20,58 +20,41 @@ use crate::*;
 
 /// All migrations that will run on the next runtime upgrade.
 pub type Migrations = (
-    // migration for added section sizes
-    pallet_gear_program::migrations::add_section_sizes::AddSectionSizesMigration<Runtime>,
-    // substrate v1.4.0
-    staking::MigrateToV14<Runtime>,
-    pallet_grandpa::migrations::MigrateV4ToV5<Runtime>,
     // move allocations to a separate storage item and remove pages_with_data field from program
     pallet_gear_program::migrations::allocations::MigrateAllocations<Runtime>,
```
Comment (breathx) [inline runtime/vara/src/migrations.rs:24]: if Im not mistaken, could be also removed?

**Review by breathx — APPROVED** [review]

File: node/service/src/rpc/mod.rs (hunk 1 above)
Comment (ukint-vs) [reply node/service/src/rpc/mod.rs:172]: deprecated

File: runtime/vara/src/integration_tests.rs (hunk 4 above)
Comment (ukint-vs) [reply runtime/vara/src/integration_tests.rs:546]: there is no ability to propose anything directly after new treasury changes. Test is unreliable.

File: runtime/common/src/apis.rs (hunk 3 above)
Comment (ukint-vs) [reply runtime/common/src/apis.rs:34]: Support is enabled

**Review by breathx — APPROVED** [review]

**Pull request merged**